edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/DeviceClientWrapper.cs

```diff
@@ -24,9 +24,15 @@ public DeviceClientWrapper(DeviceClient deviceClient)
 
         public Task AbandonAsync(string messageId) => this.underlyingDeviceClient.AbandonAsync(messageId);
 
-        public Task CloseAsync() => this.isActive.GetAndSet(false)
-            ? this.underlyingDeviceClient.CloseAsync()
-            : Task.CompletedTask;
+        public Task CloseAsync()
+        {
+            if (this.isActive.GetAndSet(false))
+            {
+                this.underlyingDeviceClient?.Dispose();
```

Comment:
Do you need the null check on underlyingClient?

Reply:
not really needed, I did same as Dispose. I can remove it, but I would like to do it in a PR that has to go to master and than backport to 1.0.9